v2/blockstore/readonly_test.go

```diff
@@ -118,6 +118,84 @@ func TestNewReadOnlyFailsOnUnknownVersion(t *testing.T) {
 	require.Nil(t, subject)
 }
 
+func TestReadOnlyAllKeysChanErrHandlerCalledOnTimeout(t *testing.T) {
+	expiredCtx, cancel := context.WithTimeout(context.Background(), -time.Millisecond)
+	t.Cleanup(cancel)
+
+	subject, err := OpenReadOnly("../testdata/sample-v1.car")
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, subject.Close()) })
+
+	// Make a channel to be able to select and block on until error handler is called.
+	errHandlerCalled := make(chan interface{})
+	expiredErrHandlingCtx := WithAsyncErrorHandler(expiredCtx, func(err error) {
+		defer close(errHandlerCalled)
+		require.EqualError(t, err, "context deadline exceeded")
+	})
+	_, err = subject.AllKeysChan(expiredErrHandlingCtx)
+	require.NoError(t, err)
+
+	// Assert error handler was called with required condition, waiting at most 3 seconds.
+	select {
+	case <-errHandlerCalled:
+		t.Log("error handler called as expected")
```

Comment:
I don't think tests should log when things work normally, unless the test is very slow to the point of needing to show progress :) Otherwise `go test -v` would tend to be very verbose